src/Orleans.Runtime/Silo/TestHooks/TestHooksSystemTarget.cs

```diff
@@ -40,60 +39,51 @@ public Task<string> GetConsistentRingProviderDiagnosticInfo()
+            return Task.FromResult(this.host.Services.GetServiceByName<IStorageProvider>(providerName) != null);
         }
 
         public Task<bool> HasStreamProvider(string providerName)
         {
             try
             {
-                this.host.Services.GetRequiredService<IStreamProviderManager>().GetStreamProvider(providerName);
-                return Task.FromResult(true);
+                return Task.FromResult(this.host.Services.GetServiceByName<IStreamProvider>(providerName) != null);
             }
             catch (KeyNotFoundException)
             {
                 return Task.FromResult(false);
             }
         }
 
-        public Task<bool> HasBoostraperProvider(string providerName)
+        public Task<ICollection<string>> GetStorageProviderNames()
         {
-            foreach (var provider in this.host.Services.GetRequiredService<BootstrapProviderManager>().GetProviders())
-            {
-                if (String.Equals(providerName, provider.Name))
-                {
-                    return Task.FromResult(true);
-                }
-            }
-            return Task.FromResult(false);
+            var storageProviderCollection = this.host.Services.GetRequiredService<IKeyedServiceCollection<string, IStorageProvider>>();
+            return Task.FromResult<ICollection<string>>(storageProviderCollection.GetServices(this.host.Services).Select(keyedService => keyedService.Key).ToArray());
         }
 
-        public Task<ICollection<string>> GetStorageProviderNames() => Task.FromResult<ICollection<string>>(this.host.Services.GetRequiredService<StorageProviderManager>().GetProviderNames().ToList());
-
-        public Task<ICollection<string>> GetStreamProviderNames() => Task.FromResult<ICollection<string>>(this.host.Services.GetRequiredService<IStreamProviderManager>().GetStreamProviders().Select(p => ((IProvider)p).Name).ToList());
+        public Task<ICollection<string>> GetStreamProviderNames()
+        {
+            var streamProviderCollection = this.host.Services.GetRequiredService<IKeyedServiceCollection<string, IStreamProvider>>();
+            return Task.FromResult<ICollection<string>>(streamProviderCollection.GetServices(this.host.Services).Select(keyedService => keyedService.Key).ToArray());
+        }
 
         public Task<ICollection<string>> GetAllSiloProviderNames()
         {
             List<string> allProviders = new List<string>();
 
-            var storageProviderManager = this.host.Services.GetRequiredService<StorageProviderManager>();
-            allProviders.AddRange(storageProviderManager.GetProviderNames());
-
-            var streamProviderManager = this.host.Services.GetRequiredService<IStreamProviderManager>();
-            allProviders.AddRange(streamProviderManager.GetStreamProviders().Select(p => p.Name));
+            var storageProviderCollection = this.host.Services.GetRequiredService<IKeyedServiceCollection<string,IStorageProvider>>();
```

Comment:
why not reuse `GetStorageProviderNames`

Reply:
those calls were unnecessarily asynchrony, but don't think that matters really.  Will use them, it's better than code duplication.